Criar script para gerar figuras de gráficos de variações de temperatura

YagoMCastro opened the pull request:
This PR introduces a script that computes the rate of temperature change over the last five years for a given country. I also translates the comments made on #7 to Portuguese.
Related to #4.

YagoMCastro committed:
Plot data

YagoMCastro committed:
Translate and add comments to portuguese

YagoMCastro committed:
Run black

leouieda commented on code/plota_dados.py:
```diff
@@ -0,0 +1,54 @@
+
+# Lê o csv ignorando comentários
+data = pd.read_csv(
+    file_path, comment="#", sep="|", names=["regiao", "variacao_temperatura"]
```
“pais” ao invés de região.

leouieda commented on code/plota_dados.py:
```diff
@@ -0,0 +1,54 @@
+# Ordena dados pela variação de temperatura
+data = data.sort_values(by="variacao_temperatura")
+
+fig, ax = plt.subplots()
```
Fazer 1 única figura com 2 subplots ao invés de duas figuras.

leouieda commented on code/plota_dados.py:
```diff
@@ -0,0 +1,54 @@
+ax.legend(title="Região")
+
+# Salva gráfico
+plt.savefig("code/img/cinco_menores_taxas")
```
1 figura só e o caminho deve ser passado na linha de comando

leouieda commented on code/plota_dados.py:
```diff
@@ -0,0 +1,54 @@
+
+ax.bar(data["regiao"].head(5), data["variacao_temperatura"].head(5), label=bar_labels)
+
+ax.set_ylabel("Variação de temperatura")
```
```suggestion
ax.set_ylabel("Taxa de variação de temperatura (°C/ano)")
```

leouieda commented on code/plota_dados.py:
```diff
@@ -0,0 +1,54 @@
+fig, ax = plt.subplots()
+
+# Gráfico das 5 menores variações
+bar_labels = data["regiao"].head(5)
```
Não precisa de labels já que o eixo x é o país.

leouieda commented on code/plota_dados.py:
```diff
@@ -0,0 +1,54 @@
+# Gráfico das 5 menores variações
+bar_labels = data["regiao"].head(5)
+
+ax.bar(data["regiao"].head(5), data["variacao_temperatura"].head(5), label=bar_labels)
```
```suggestion
ax.bar(data["regiao"].head(5), data["variacao_temperatura"].head(5))
```

leouieda commented on code/plota_dados.py:
```diff
@@ -0,0 +1,54 @@
+ax.bar(data["regiao"].head(5), data["variacao_temperatura"].head(5), label=bar_labels)
+
+ax.set_ylabel("Variação de temperatura")
+ax.set_title("Menores Variações de Temperatura")
```
```suggestion
ax.set_title("Cinco maiores")
```

leouieda commented on code/plota_dados.py:
```diff
@@ -0,0 +1,54 @@
+
+ax.set_ylabel("Variação de temperatura")
+ax.set_title("Menores Variações de Temperatura")
+ax.legend(title="Região")
```
```suggestion
```

leouieda commented on code/img/cinco_maiores_taxas.png:
Remover

leouieda commented on code/img/cinco_menores_taxas.png:
Remover

leouieda committed:
Adiciona um script para calcular variação de todos os países (#9)

O script roda o código Python e gera os resultados para todos os
arquivos. Não sei se vale a pena fazer isso. Vai depender do andamento
da aula. Qualquer coisa o for pode ser escrito direto no Makefile.

YagoMCastro committed:
Change from "Variação de Temperatura" to "Taxa de variação de temperatura""

Co-authored-by: Leonardo Uieda <leo@uieda.com>

YagoMCastro committed:
Remove title from figure

Co-authored-by: Leonardo Uieda <leo@uieda.com>

YagoMCastro committed:
Update title

Co-authored-by: Leonardo Uieda <leo@uieda.com>

YagoMCastro committed:
Remove label from bar graphic figure

Co-authored-by: Leonardo Uieda <leo@uieda.com>

YagoMCastro committed:
Remove saved images and some minor fixes to the figure

YagoMCastro committed:
Change from regiao to pais

leouieda committed:
Adiciona um paper em LaTeX e um Makefile (#11)

O paper usa variáveis e cria um artigo padrão com citações. O Makefile
calcula os resultados rodando o código Python e Bash gera o PDF. O
Makefile também gera arquivos .tex com variáveis para incluir no texto
(um com bash mesmo e outro com um novo script Python). Modifiquei o
script de análise em Python para imprimir com separação por vírgula e
com `""` entorno do nome. Assim dá para salvar as coisas como CSV e
carregar no pandas sem nenhum comando especial.

YagoMCastro committed:
Update figure

YagoMCastro committed:
Update comments

leouieda committed:
Usa código para baixar os dados

Script em Python usa o Pooch para baixar os dados.

leouieda committed:
Simplifica o LaTeX

Remove alguns pacotes legais mas que não são necessários.

leouieda commented on code/plota_dados.py:
```diff
@@ -0,0 +1,47 @@
+menores variações de temperatura.
+
+O caminho até o arquivo com os dados em formato CSV deve ser o primeiro
+argumento da linha de comando. As figuras são armazenadas em code/img/.
```
```suggestion
argumento da linha de comando.
```

leouieda committed:
Apply suggestions from code review

leouieda committed:
Adiciona o plot no latex

leouieda committed:
Merge branch 'main' into plot
